### PR TITLE
[WIP] Add Bank example

### PR DESCRIPTION
**Pull request by boekkooi-fresh**
In order to give interested users a better example and some code the bank example is added.
This example project uses the aggregate, stream projector and aggregate projector.

### CODE REVIEW

File: example/bank/README.md [hunk 1]
```diff
@@ -0,0 +1,23 @@
+## Bank Example
+
+Welcome to the GoEngine Bank. 
+
+This is a basic example of how GoEngine can be used. This application consists of the following commands:
+- [API](cmd/api) The banks HTTP API, here bank account can be opened, money can be deposited and withdrawn.
+- [report-bank](cmd/report-bank) The banks general reported, this will report the total balance across all accounts and the total amount of bank accounts.
```
Comment (sergiughf) [inline example/bank/README.md]: Probably `banks general report,..` ? and not `reported`¿

File: example/bank/README.md [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+- [report-account](cmd/report-account) The account reporter keeps track of the average accounts debit and credit information.  
+
+### Running the example
+
+In order to run the example you need a postgres DB. The provided [docker-compose.yml](docker-compose.yml) provides this.
```
Comment (sergiughf) [inline example/bank/README.md]: Maybe let's not repeat `provides` so much and rephrase this as `The provided bla bla bla contains this.` Wdyt?:)

File: example/bank/README.md [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+### Running the example
+
+In order to run the example you need a postgres DB. The provided [docker-compose.yml](docker-compose.yml) provides this. 
+
+Once you have your DB server up and running you need to set the DSN using the `POSTGRES_DSN` environment variable.
+```bash   
+export POSTGRES_DSN="postgres://goengine:goengine@localhost/goengine-bank?sslmode=disable&client_encoding=UTF8"
+```
+
+Now you can start once of the applications.
```
Comment (sergiughf) [inline example/bank/README.md]:
```suggestion
Now you can start one of the applications.
```

File: example/bank/cmd/api/account.go [hunk 4]
```diff
@@ -0,0 +1,240 @@
...
+
+	w.Header().Add("Content-Type", "application/json")
+	w.WriteHeader(http.StatusOK)
+	if _, err := w.Write(body); err != nil {
+		a.logger.With(zap.Error(err)).Error("failed to write response")
+	}
+}
+
+func (a *accountHandler) AccountAverages(w http.ResponseWriter, r *http.Request) {
+	var (
```
Comment (sergiughf) [inline example/bank/cmd/api/account.go]:
```suggestion
	var debit, credit uint
```

You can have this on one line :)

File: example/bank/cmd/api/account.go [hunk 5]
```diff
@@ -0,0 +1,240 @@
...
+	w.WriteHeader(http.StatusOK)
+	if _, err = w.Write(body); err != nil {
+		a.logger.With(zap.Error(err)).Error("failed to write response")
+	}
+}
+
+func (a *accountHandler) AccountDeposit(w http.ResponseWriter, r *http.Request) {
+	amount, err := a.loadAmountFromPost(w, r)
+	if err != nil {
+		return
```
Comment (sergiughf) [inline example/bank/cmd/api/account.go]: No error message to display?

File: example/bank/cmd/api/account.go [hunk 6]
```diff
@@ -0,0 +1,240 @@
...
+	}
+
+	bankAccount := a.loadAccount(w, r)
+	if bankAccount == nil {
+		return
```
Comment (sergiughf) [inline example/bank/cmd/api/account.go]: No error message to display?

File: example/bank/cmd/api/account.go (hunk 5 above)
Comment (boekkooi-fresh) [reply example/bank/cmd/api/account.go]: The error is done by `loadAmountFromPost` I didn't want to duplicate but it may indeed be a little confusing.... Let me see if I can improve this.

File: example/bank/cmd/api/account.go (hunk 6 above)
Comment (boekkooi-fresh) [reply example/bank/cmd/api/account.go]: The error is done by `loadAccount` I didn't want to duplicate but it may indeed be a little confusing.... Let me see if I can improve this.

File: example/bank/cmd/api/main.go [hunk 7]
```diff
@@ -0,0 +1,104 @@
...
+	failOnError(err)
+
+	bankAccountRepo, err := domain.NewBankAccountRepository(eventStore, config.EventStoreStreamName)
+	failOnError(err)
+
+	router := chi.NewRouter()
+	router.Get("/", listRoutes(router, logger))
+	router.Get("/report/", bankReportsHandler(db, logger))
+	router.Mount("/debug", middleware.Profiler())
+	router.Mount("/account", newAccountHandler(db, bankAccountRepo, logger).HTTPHandler())
```
Comment (spetrashov) [inline example/bank/cmd/api/main.go]: This one stands a little bit out. Why don't you return `*chi.Mux` straight away instead of calling `newAccountHandler(...). HTTPHandler()`?

File: example/bank/config/goengine.go [hunk 8]
```diff
@@ -0,0 +1,48 @@
...
+	"time"
+
+	"github.com/hellofresh/goengine"
+	"github.com/hellofresh/goengine/example/bank/domain"
+	"github.com/hellofresh/goengine/extension/pq"
+	"github.com/hellofresh/goengine/strategy/json"
+	"github.com/hellofresh/goengine/strategy/json/sql/postgres"
+)
+
+// NewGoEngineManager return a confiured SingleStreamManager for the application
```
Comment (spetrashov) [inline example/bank/config/goengine.go]:
```suggestion
// NewGoEngineManager return a configured SingleStreamManager for the application
```

File: example/bank/cmd/api/account.go [hunk 9]
```diff
@@ -0,0 +1,242 @@
...
+	}
+
+	accountID := r.Context().Value(contextKeyBankAccountID).(aggregate.ID)
+	bankAccount, err := a.bankAccountRepo.Get(r.Context(), accountID)
+	if err != nil {
+		a.renderAccountError(w, accountID, err)
+		return
+	}
+
+	if err := bankAccount.Deposit(amount); err != nil {
```
Comment (puellanivis) [inline example/bank/cmd/api/account.go]:
```suggestion
	if err := bankAccount.Withdraw(amount); err != nil {
```
